dcicutils/secrets_utils.py

```diff
@@ -53,12 +58,73 @@ def assume_identity():
         if 'SecretString' in get_secret_value_response:
-            identity = json.loads(get_secret_value_response['SecretString'])
+            identity_secrets = json.loads(get_secret_value_response['SecretString'])
         else:
             raise Exception('Got unexpected response structure from boto3')
-    if not identity:
+    if not identity_secrets:
         raise Exception('Identity could not be found! Check the secret name.')
-    return identity
+    return identity_secrets
+
+
+assume_identity = get_identity_secrets  # assume_identity is deprecated. Please rewrite
+
+
+@contextlib.contextmanager
+def assumed_identity_if(only_if, *, identity_kind=GLOBAL_APPLICATION_CONFIGURATION, only_if_missing=None,
+                        require_identity=False):
+    with assumed_identity(identity_kind=identity_kind, only_if=only_if, only_if_missing=only_if_missing,
+                          require_identity=require_identity):
+        yield
+
+
+@contextlib.contextmanager
+def assumed_identity(*, identity_kind=GLOBAL_APPLICATION_CONFIGURATION, only_if=True, only_if_missing=None,
+                     require_identity=False):
+    """
+    Assumes a given identity in a context.
+
+    NOTE: This assumes that global_env_bucket is in the GAC and not otherwise,
+          so it tests
+
+    :param identity_kind: The name of the identity to assume (default 'IDENTITY')
+    :param only_if: Whether to try assuming identity at all (default True)
+    :param only_if_missing: The name of an environment variable that would only be in the GAC.
+       Load the GAC only if this variable is not set. Default is None, meaning load unconditionally.
+    :param require_identity: Whether to raise an error if the identity_kind environment variable is not bound.
+
+    """
+    if only_if and (not only_if_missing or not os.environ.get(only_if_missing)):
+        identity_name = get_identity_name(identity_kind=identity_kind)
+        if identity_name:
+            secrets = get_identity_secrets(identity_kind=identity_kind)
+            if only_if_missing not in secrets:
+                raise RuntimeError(f"No {only_if_missing} was found where expected"
+                                   f" in {identity_kind} secrets at {identity_name}.")
+            with override_environ(**secrets):
+                yield
+                return  # Nothing to do after that
+
+        elif require_identity:
+            raise RuntimeError(f"An identity was not defined in environment variable {identity_kind}.")
+
+    yield
+
+
+def apply_identity(identity_kind=GLOBAL_APPLICATION_CONFIGURATION):
+    """
+    Assumes an identity globally.
+
+    Since this is assumed to be an initial startup action, we assume it's not already been done and don't
+    offer the only_if_missing mechanism that the assumed_identity context manager offers.
+
+    """
+    gac = os.environ.get(identity_kind)
```

Comment:
Couldn't/shouldn't this be a call to: get_identity_name(identity_kind) ... ? (And variable named identity_name rather than gac?)